### PR TITLE
#17975 : Improving the error messages that come from the Push Now act…

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
…ionlet, and some minor code formatting and Javadoc.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java [hunk 1]
```diff
@@ -42,94 +58,78 @@ public String getName() {
...
-            Contentlet ref = processor.getContentlet();
-
-            boolean _contentPushNeverExpire = true;
-
-            boolean forcePush = ("true".equals(params.get("force").getValue())) ? true : false;
-            String envrions = params.get("environment").getValue();
-            if (envrions == null) {
-                Logger.error(this.getClass(), "There are no environments set to push to");
+            final Contentlet contentlet = processor.getContentlet();
+            final boolean forcePush = ("true".equals(params.get(PARAM_FORCE_PUSH).getValue())) ? Boolean.TRUE : Boolean.FALSE;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4986076)

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java [hunk 2]
```diff
@@ -42,94 +58,78 @@ public String getName() {
...
-            String[] whereToSend = envrions.split(",");
-
-            List<Environment> envsToSendTo = new ArrayList<Environment>();
-            List<Environment> permissionedEnv = new ArrayList<Environment>();
-            List<Environment> finalEnvs = new ArrayList<Environment>();
-
-
+            final String[] whereToSend = environments.split(ENVIRONMENT_DELIMITER);
+            final List<Environment> envsToSendTo = new ArrayList<>();
+            final List<Environment> permissionedEnv = new ArrayList<>();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java]: Codacy Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4986076)

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java [hunk 3]
```diff
@@ -42,94 +58,78 @@ public String getName() {
...
-                for(Role r: roles){
+                }
+            } else {
+                for (final Role role : roleList){
                     try {
-                        permissionedEnv.addAll(APILocator.getEnvironmentAPI().findEnvironmentsByRole(r.getId()));
-                    } catch (Exception e) {
-                        Logger.error(PushNowActionlet.class, e.getMessage());
+                        permissionedEnv.addAll(APILocator.getEnvironmentAPI().findEnvironmentsByRole(role.getId()));
+                    } catch (final Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java]: Codacy Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4986076)

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java [hunk 4]
```diff
@@ -42,94 +58,78 @@ public String getName() {
...
-
-            List<Environment> envsToSendTo = new ArrayList<Environment>();
-            List<Environment> permissionedEnv = new ArrayList<Environment>();
-            List<Environment> finalEnvs = new ArrayList<Environment>();
-
-
+            final String[] whereToSend = environments.split(ENVIRONMENT_DELIMITER);
+            final List<Environment> envsToSendTo = new ArrayList<>();
+            final List<Environment> permissionedEnv = new ArrayList<>();
+            final List<Environment> finalEnvs = new ArrayList<>();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java]: Codacy Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4986076)

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java]: Codacy Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4986076)

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java [hunk 5]
```diff
@@ -42,94 +58,78 @@ public String getName() {
...
-
-            Contentlet ref = processor.getContentlet();
-
-            boolean _contentPushNeverExpire = true;
-
-            boolean forcePush = ("true".equals(params.get("force").getValue())) ? true : false;
-            String envrions = params.get("environment").getValue();
-            if (envrions == null) {
-                Logger.error(this.getClass(), "There are no environments set to push to");
+            final Contentlet contentlet = processor.getContentlet();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java]: Codacy Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4986076)

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java [hunk 6]
```diff
@@ -42,94 +58,78 @@ public String getName() {
...
             // make sure the user has permissions to push
-            boolean isAdmin = APILocator.getUserAPI().isCMSAdmin(processor.getUser());
-            List<Role> roles = APILocator.getRoleAPI().loadRolesForUser(processor.getUser().getUserId(),true);
-            if(isAdmin){
-                List<Environment> app = APILocator.getEnvironmentAPI().findEnvironmentsWithServers();
-                for(Environment e:app)
+            final boolean isAdmin = APILocator.getUserAPI().isCMSAdmin(processor.getUser());
+            final List<Role> roleList = APILocator.getRoleAPI().loadRolesForUser(processor.getUser().getUserId(),true);
+            if (isAdmin) {
+                List<Environment> environmentList = APILocator.getEnvironmentAPI().findEnvironmentsWithServers();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java]: Codacy Issue found: [Local variable 'environmentList' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4986076)

**Review by dsilvam — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java [hunk 7]
```diff
@@ -42,94 +59,78 @@ public String getName() {
...
-                for(Role r: roles){
+                }
+            } else {
+                for (final Role role : roleList){
                     try {
-                        permissionedEnv.addAll(APILocator.getEnvironmentAPI().findEnvironmentsByRole(r.getId()));
-                    } catch (Exception e) {
-                        Logger.error(PushNowActionlet.class, e.getMessage());
+                        permissionedEnv.addAll(APILocator.getEnvironmentAPI().findEnvironmentsByRole(role.getId()));
+                    } catch (final Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java]: Codacy Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4986076)

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java [hunk 8]
```diff
@@ -42,94 +59,78 @@ public String getName() {
...
-
-            Contentlet ref = processor.getContentlet();
-
-            boolean _contentPushNeverExpire = true;
-
-            boolean forcePush = ("true".equals(params.get("force").getValue())) ? true : false;
-            String envrions = params.get("environment").getValue();
-            if (envrions == null) {
-                Logger.error(this.getClass(), "There are no environments set to push to");
+            final boolean forcePush = "true".equals(params.get(PARAM_FORCE_PUSH).getValue()) ? Boolean.TRUE : Boolean.FALSE;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java]: Codacy Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4986076)

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java [hunk 9]
```diff
@@ -1,12 +1,18 @@
...
+import java.util.Map;
+
 import com.dotcms.publisher.bundle.bean.Bundle;
 import com.dotcms.publisher.business.DotPublisherException;
 import com.dotcms.publisher.business.PublisherAPI;
 import com.dotcms.publisher.environment.bean.Environment;
 import com.dotmarketing.business.APILocator;
 import com.dotmarketing.business.Role;
 import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java]: Codacy Issue found: [Avoid unused imports such as 'com.dotmarketing.exception.DotSecurityException'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4986076)

**Review by jdotcms — APPROVED** [review]

**Comment by jcastro-dotcms** [comment]
Fix will be taken care of by another PR

**Pull request closed without merging**